apps/io-func-wallet-solution-backend/src/infra/federation-entity/config.ts

```diff
@@ -0,0 +1,28 @@
+import * as RE from "fp-ts/lib/ReaderEither";
+import { sequenceS } from "fp-ts/lib/Apply";
+import { pipe } from "fp-ts/lib/function";
+
+import { readFromEnvironment } from "../env";
+import { FederationEntityMetadata } from "../../entity-configuration";
+import { validate } from "../../validation";
+
+export const getFederationEntityConfigFromEnvironment: RE.ReaderEither<
+  NodeJS.ProcessEnv,
+  Error,
+  FederationEntityMetadata
+> = pipe(
+  sequenceS(RE.Apply)({
+    basePath: readFromEnvironment("FederationEntityBasePath"),
+    organizationName: readFromEnvironment("FederationEntityOrganizationName"),
+    homePageUri: readFromEnvironment("FederationEntityHomepageUri"),
+    policyUri: readFromEnvironment("FederationEntityPolicyUri"),
+    tosUri: readFromEnvironment("FederationEntityTosUri"),
+    logoUri: readFromEnvironment("FederationEntityLogoUri"),
```

Comment:
**issue(complexity)**: I see a lot of complexity I don't think it's needed. Here we are duplicating configuration keys from `FederationEntityMetadata`, with all the effort of keeping definitions aligned. I see why we are doing that - because the name of a configuration key may not be the same of the environment variable. Nevertheless, I question if it's worth the cognitive effort to understand it.

I had to bounce 3 different files to just understand which environment parameters are expected by the application - and I'm not done yet. We should really consider to keep our codebase simple.

Reply:
> A configuration file is needed for each infrastructure component to not be dependent on an environment within the business logic

This is for sure a good pattern. I doubt it's a *need* for our architecture - I think it's something we do because we find it useful.
What I argue is that it makes the code less _stupid_, and the complexity of the abstraction overweights the value it brings.
Why cannot we simply have a `config.ts` file, in which we can find all the environment variables mapped to the application configuration object? Just like we do in legacy [legacy](https://github.com/pagopa/io-functions-app/blob/master/utils/config.ts) as well as [new](https://github.com/pagopa/io-services-cms/blob/master/apps/io-services-cms-webapp/src/config.ts) codebases.

----

> All the environment parameters required by the application can be found in `local.settings.json.example`

I discourage this approach, as example files always end up being off-sync with both the code and documentation (unless we automatically check they are). 
IMHO it's preferable to keep code as source of truth.

Reply:
Ok I understand. 

I've tried moving the federation entity configuration directly into the main `config.ts` by deleting the whole folder on infra. It was too overkilling!

As far as the crypto configuration is concerned, I preferred to leave it within infra since the infrastructure files (`signer.ts`) depend on the configuration. Furthermore, the advantage of this implementation is that in case it is necessary to add new parameters to the configuration, it is sufficient to modify a single file directly in infra.

Reply:
Ok, I've moved all the configs under one single file in `src/app/config.ts` !